### PR TITLE
monad-raptorcast: Reorder post-proof header fields to improve alignment

### PR DESCRIPTION
**Pull request by buytenh**
Without the changes in this PR, the part of the RaptorCast header
after the Merkle proof section looks like this:

```
             0        4        8        c
    0x0000:  xx                                   chunk_merkle_leaf_idx
               xxxxxx xxxxxxxx xxxxxxxx xxxxxxxx  chunk_recipient
    0x0010:  xxxxxxxx xx
                        xxxx                      chunk_id
                            xx xx                 chunk_len
                                 xxxxxx xx......  data
```

Note the lack of proper alignment of several of the fields, and how the
data field has a length that is congruent to 3 mod 4.

This PR moves the chunk_merkle_leaf_idx byte to be after the
chunk_recipient field, and removes the chunk_len field (since we can
infer the chunk length from the length of the packet) and replaces it
with a single padding byte.  This makes the layout look like this:

```
             0        4        8        c
    0x0000:  xxxxxxxx xxxxxxxx xxxxxxxx xxxxxxxx  chunk_recipient
    0x0010:  xxxxxxxx
                      xx                          chunk_merkle_leaf_idx
                        xx                        reserved
                          xxxx                    chunk_id
                               xxxxxxxx ........  data
```

This looks a bit better, and this way, all fields are naturally aligned,
and the data field is a multiple of 4 bytes long.

### CODE REVIEW

File: monad-raptorcast/src/udp.rs [hunk 1]
```diff
@@ -228,7 +228,6 @@ impl<ST: CertificateSignatureRecoverable> UdpState<ST> {
 //     chunk_merkle_leaf_idx: u8,
```
Comment (omegablitz) [inline monad-raptorcast/src/udp.rs]: Forgot to move this?

File: monad-raptorcast/src/udp.rs [hunk 2]
```diff
@@ -620,32 +617,23 @@ where
...
-    let cursor_payload_len = split_off(2)?;
-    let payload_len = u16::from_le_bytes(
-        cursor_payload_len
-            .as_ref()
-            .try_into()
-            .expect("u16 is 2 bytes"),
-    );
-    if payload_len == 0 {
-        // handle the degenerate case
-        return Err(MessageValidationError::TooShort);
```
Comment (omegablitz) [inline monad-raptorcast/src/udp.rs:645]: Is this condition handled somewhere else now?Just checking that a zero payload won't break; specifically worried about this:
https://github.com/monad-crypto/monad-bft/blob/316f00f46a8ae739b3edbaf6cf8219140245b480/monad-raptorcast/src/udp.rs#L144-L146

File: monad-raptorcast/src/udp.rs (hunk 1 above)
Comment (buytenh) [reply monad-raptorcast/src/udp.rs]: Thanks, fixed.

File: monad-raptorcast/src/udp.rs (hunk 2 above)
Comment (buytenh) [reply monad-raptorcast/src/udp.rs:645]: Good catch -- I added back the length check:
```
     let cursor_payload = cursor;
+    if cursor_payload.is_empty() {
+        // handle the degenerate case
+        return Err(MessageValidationError::TooShort);
+    }
```

File: monad-raptorcast/src/udp.rs [hunk 3]
```diff
@@ -225,10 +225,9 @@ impl<ST: CertificateSignatureRecoverable> UdpState<ST> {
 //
 //     merkle_proof: Vec<[u8; 20]>,
 //
-//     chunk_merkle_leaf_idx: u8,
 //     chunk_recipient: [u8; 20],
+//     chunk_merkle_leaf_idx: u8,
```
Comment (omegablitz) [inline monad-raptorcast/src/udp.rs:229]: one last nit: could you also add the `reserved: u8` here for completeness?

Comment (buytenh) [reply monad-raptorcast/src/udp.rs:229]: done

**Pull request merged**